api/app/Http/Controllers/ChildEntityController.php

```diff
@@ -52,16 +74,19 @@ public function getAll($id)
      */
     public function getOne($id, $childId)
     {
-        $model = $this->getParentModel($id);
+        $this->validateId($id, $this->getParentModel()->getKeyName(), $this->validateParentRequestRule);
+        try {
+            $model =  $this->getParentEntityById($id);
```

Comment:
The `$this->getParentEntityById` method should throw a `notFoundException` directly as it is a controller function. This would simplify many of these controller methods.